mzLib/UsefulProteomicsDatabases/ProteinDbWriter.cs

```diff
@@ -0,0 +1,136 @@
+                        writer.WriteAttributeString("position", proteolysisProduct.OneBasedEndPosition.ToString());
+                        writer.WriteEndElement();
+                        writer.WriteEndElement();
+                        writer.WriteEndElement();
+                    }
+                    foreach (var ye in protein.OneBasedPossibleLocalizedModifications.OrderBy(b => b.Key))
+                    {
+                        foreach (var nice in ye.Value)
+                        {
+                            writer.WriteStartElement("feature");
+                            writer.WriteAttributeString("type", "modified residue");
+                            writer.WriteAttributeString("description", nice.id);
+                            writer.WriteStartElement("location");
+                            writer.WriteStartElement("position");
+                            writer.WriteAttributeString("position", ye.Key.ToString(CultureInfo.InvariantCulture));
+                            writer.WriteEndElement();
+                            writer.WriteEndElement();
+                            writer.WriteEndElement();
+                        }
+                    }
+                    if (Mods.ContainsKey(protein.Accession))
+                        foreach (var ye in Mods[protein.Accession].OrderBy(b => b.Item1))
+                        {
+                            writer.WriteStartElement("feature");
+                            writer.WriteAttributeString("type", "modified residue");
+                            writer.WriteAttributeString("description", ye.Item2.id);
+                            writer.WriteStartElement("location");
+                            writer.WriteStartElement("position");
+                            writer.WriteAttributeString("position", ye.Item1.ToString(CultureInfo.InvariantCulture));
+                            writer.WriteEndElement();
+                            writer.WriteEndElement();
+                            writer.WriteEndElement();
+                        }
+
+                    writer.WriteStartElement("sequence");
+                    writer.WriteAttributeString("length", protein.Length.ToString(CultureInfo.InvariantCulture));
+                    writer.WriteString(protein.BaseSequence);
+                    writer.WriteEndElement();
+
+                    writer.WriteEndElement();
+                }
+
+                writer.WriteEndElement();
+                writer.WriteEndDocument();
+            }
+        }
+
+        public static void WriteFastaDatabase(List<Protein> proteinList, string outputFileName)
+        {
+            using (StreamWriter writer = new StreamWriter(outputFileName))
+            {
+                foreach(Protein protein in proteinList)
+                {
+                    string header = protein.FullName.Contains("|") ?
+                        protein.Accession + "|" + protein.FullName :
```

Comment:
So what I don't like about this is the idea that our code can output fasta headers in two different formats, depending on some minute detail in the protein full name. 
Why can't we pick one or the other? Space or | for outputting?